src/robin_dict.jl

```diff
@@ -0,0 +1,189 @@
+        h[p.first] = p.second
+    end
+    return h
+end
+
+RobinDict() = RobinDict{Any,Any}()
+RobinDict(kv::Tuple{}) = RobinDict()
+
+RobinDict(ps::Pair{K,V}...) where {K,V} = RobinDict{K,V}(ps)
+RobinDict(ps::Pair...) = RobinDict(ps)
+
+# default hashing scheme used by Julia
+hashindex(key, sz) = (((hash(key)%Int) & (sz-1)) + 1)::Int
+
+# insert algorithm
+function rh_insert!(h::RobinDict{K, V}, key::K, val::V) where {K, V}
+    # table full
+    if h.count == length(h.keys)
+        return -1
+    end
+    ckey, cval, cdibs = key, val, 0
+    sz = length(h.keys)
+    index = hashindex(ckey, sz)
+    @inbounds while true
+    	if (h.slots[index] == 0x0) || (h.slots[index] == 0x1 && h.keys[index] == ckey)
+    		break
+    	end
+        if h.dibs[index] < cdibs
+            h.vals[index], cval = cval, h.vals[index]
+            h.keys[index], ckey = ckey, h.keys[index]
+            h.maxprobe = max(h.maxprobe, cdibs)
+            h.dibs[index], cdibs = cdibs, h.dibs[index]
+        end
+        cdibs += 1
+        h.totalcost += 1
+        index = (index & (sz - 1)) + 1
+    end
+    # println("Successfully inserted at $index")
+    @inbounds if h.slots[index] == 0x0
+    	h.count += 1
+    end
+    @inbounds h.slots[index] = 0x1
+    @inbounds h.vals[index] = cval
+    @inbounds h.keys[index] = ckey
+    @inbounds h.dibs[index] = cdibs
+    h.totalcost += 1
+    h.maxprobe = max(h.maxprobe, cdibs)
+    if h.idxfloor == 0
+        h.idxfloor = index
+    else
+        h.idxfloor = min(h.idxfloor, index)
+    end
+    return index
+end
+
+@propagate_inbounds isslotempty(h::RobinDict{K, V}, i) where {K, V} = h.slots[i] == 0x0
+@propagate_inbounds isslotfilled(h::RobinDict{K, V}, i) where {K, V} = h.slots[i] == 0x1
+@propagate_inbounds isslotdeleted(h::RobinDict{K, V}, i) where {K, V} = h.slots[i] == 0x2
+
+function setindex!(h::RobinDict{K,V}, v0, key0) where {K, V}
+    key = convert(K, key0)
+    isequal(key, key0) || throw(ArgumentError("$key0 is not a valid key for type $K"))
+    _setindex!(h, key, v0)
+end
+
+function _setindex!(h::RobinDict{K,V}, key::K, v0) where {K, V}
+    v = convert(V, v0)
+    index = rh_insert!(h, key, v)
+    if index > 0
+        # println("Successfully inserted at $index")
+    else
+        throw(error("Dictionary table full"))
+    end
+    h
+end
+
+isempty(d::RobinDict) = (d.count == 0)
+length(d::RobinDict) = d.count
+
+function empty!(h::RobinDict{K,V}) where {K, V}
+    fill!(h.slots, 0x0)
+    sz = length(h.slots)
+    empty!(h.dibs)
+    empty!(h.keys)
+    empty!(h.vals)
+    resize!(h.keys, sz)
+    resize!(h.vals, sz)
+    resize!(h.dibs, sz)
+    h.count = 0
+    h.maxprobe = 0
+    h.totalcost = 0
+    h.idxfloor = 0
+    return h
+end
+ 
+function rh_search(h::RobinDict{K, V}, key::K) where {K, V}
+	sz = length(h.keys)
+	index = hashindex(key, sz)
+	cdibs = 0
+	while true
+		if h.slots[index] == 0x0
+			return -1
+		elseif cdibs > h.dibs[index]
+			return -1
+		elseif h.keys[index] == key 
```

Comment:
I'm not sure how to configure you editor,
but I can tell you there are definately tabs on this line.